### ISSUE TITLE
Feature: filename formatting

### ISSUE THREAD

**Issue by ketexon** [it-is-wednesday/cute-sway-recorder#36]
# Filename Formatting

Filename formatting, similar to OBS's `%CCYY-%MM-%DD %hh-%mm-%ss`, would be useful for sorting files by name and automatic collision avoidance.

# Potential Implementations

If we only want date/time formatting, we can use the syntax of [`strftime`](https://docs.python.org/3/library/datetime.html#strftime-strptime-behavior) (`%Y-%m-%d %H-%M-%S.mp4`), though I think we should keep the ability to have random 5 ascii digits like the current behavior.

If we want to keep the option for a random 5 ascii characters, I think [template strings](https://docs.python.org/3.7/library/string.html#template-strings) would be the "best" option, since it can look the same as `strftime`, unlike `%` formatting and `str.format`. We could use `%` as a delimiter to match `strftime` and use `%id` as the placeholder. The only potential edge case would be if `strftime` implements `%i`, but we can handle that when it happens ;)

# Considerations

If we allow a random 5 ascii characters as a template placeholder, this makes the "Random name" button a little less useful, and maybe we should consider removing it (or replacing it with "Default name"). However, it could be useful if the user wants to know where the file will be saved before starting recording. 

It might be a good idea for accessibility to show the realized filename after/before a recording is finished. Eg. a status line at the bottom that says `Saved to "~/Videos/2024-46-07/27/24 14-46-02.mp4"`. This *could* also be the same place that displays #23 .

**Comment by ketexon** [comment]
If the `strftime` + `string.Template` with `%` delimiter and `%id` behavior sounds good, let me know and I'll start working on it.

**Comment by it-is-wednesday** [comment]
Hi! Thanks so much for getting discussions running here. You're a godsend.

Frankly I'm not sure what you meant by mentioning template strings: this sounds like logic that should be implemented entirely in `strftime`, no? I didn't completely understand the logic behind your Template-based solution. Like, I didn't get neither the solution itself nor the reason to consider something other than `strftime`. 

Can you demonstrate what the `Template`-based implementation would look like?

**Comment by ketexon** [comment]
> Hi! Thanks so much for getting discussions running here. You're a godsend.
> 
> Frankly I'm not sure what you meant by mentioning template strings: this sounds like logic that should be implemented entirely in `strftime`, no? I didn't completely understand the logic behind your Template-based solution. Like, I didn't get neither the solution itself nor the reason to consider something other than `strftime`.
> 
> Can you demonstrate what the `Template`-based implementation would look like?

Oh my bad I realize I wrote little too much hehe
I meant just having an ability have have like 5 random letters in the filename: `~/Videos/%id.mp4` would generate (eg.) `~/Videos/jfab4.mp4`.

I think now though that this would belong in its own issue and is not really a priority so maybe just `strftime` for this issue unless you think otherwise :3

**Comment by it-is-wednesday** [comment]
Yeah just strftime would be sufficient. I guess there would be no need to add extra random letters except coolness factor
